framework/behaviors/AttributeBehavior.php

```diff
@@ -118,6 +123,10 @@ public function evaluateAttributes($event)
             foreach ($attributes as $attribute) {
                 // ignore attribute names which are not string (e.g. when set by TimestampBehavior::updatedAtAttribute)
                 if (is_string($attribute)) {
+                    // skip attribute update when it is not empty and the corresponding mode is enabled
+                    if ($this->skipUpdateOnFilled && !empty($this->owner->$attribute)) {
```

Comment:
@dynasource 

> this statement should be formulated in such way that a comment is not necessary.

It's not necessary, removed it.

> can you use a native function instead of empty?

Refactored that sub-condition to the `$this->owner->$attribute !== ''`.

> a check on $event->name is missing, like with $skipUpdateOnClean. This makes it confusing an less readable and error prone.

I can't get it what is the reason for checking the event name while evaluating each attribute? Can you elaborate please?